tuxemon/npc.py

```diff
@@ -847,5 +847,79 @@ def alter_item_quantity(
 
         return success
 
+    def can_buy_item(self, item_slug: str, qty: int, unit_price: int) -> bool:
+        current_money = self.game_variables["money"]
+        if current_money:
+            return current_money >= qty * unit_price
+        # If no 'money' variable, must be an NPC. Always allow buying:
+        return True
+
+    def can_sell_item(self, item_slug: str, qty: int, unit_price: int) -> bool:
+        current_item = self.inventory.get(item_slug)
+        if item_slug:
+            current_qty = current_item["quantity"]
+            return current_qty >= qty or current_item["infinite"]
+        # If they don't have the item, then they can't sell the item:
+        return False
+
+    def buy_item(
+        self, session: Session, item_slug: str, qty: int, unit_price: int
+    ) -> None:
+        """Decreases current money, and adds the items to the inventory.
+
+        Raises an exception if there's not enough money to pay the price."""
+        if self.game_variables:
```

Comment:
small suggestion, but not needed.  this check for money, quantity, and price is essentially the same thing as `can_sell_item`.  you could consider using that here as a check instead of implementing it again here.